tests/cli-inst-interactive.rs

```diff
@@ -91,6 +91,42 @@ fn smoke_test() {
     });
 }
 
+// validate host triple against a whitelist to support rustup-init
+#[test]
+fn validate_host_triple() {
+    let host_triple = &this_host_triple();
```

Comment:
don't test this_host_triple() - it is a test helper. Test from_build() - that is the function that is confused at source. We can't use from_host_or_build, as that will succeed during cross_builds, whereas from_build will return the bad data during cross_builds.

Validate it using ```PartialTargetTriple::new(&from_build())``` and we should have all we need to make these builds of rustup fail during build time.